File: app/src/main/java/example/develop/davidoh/java_android_mvp_example/data/source/flickr/FlickrDataSource.java
package example.develop.davidoh.java_android_mvp_example.data.source.flickr;

import java.util.List;

import example.develop.davidoh.java_android_mvp_example.data.PhotoItem;
import example.develop.davidoh.java_android_mvp_example.data.PhotoResponse;

public interface FlickrDataSource {
    interface LoadImageCallback {
        void onImageLoaded(int page, List<PhotoItem> items);
        void onDataNotAvailable();
        void onLoadFail(int code, String message);
    }

    void getRecentPhoto(int page, int perPage, LoadImageCallback loadImageCallback);

    void getSearchPhoto(String keyword, int page, int perPage, LoadImageCallback loadImageCallback);
}

File: app/src/main/java/example/develop/davidoh/java_android_mvp_example/data/source/flickr/FlickrDataSource.kt
package example.develop.davidoh.java_android_mvp_example.data.source.flickr

import example.develop.davidoh.java_android_mvp_example.data.PhotoItem
import example.develop.davidoh.java_android_mvp_example.data.PhotoResponse
import retrofit2.Call

interface FlickrDataSource {
    fun getRecentPhoto(page: Int, perPage: Int): Call<PhotoResponse>

    fun getSearchPhoto(keyword: String, page: Int, perPage: Int): Call<PhotoResponse>
}
